src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8610,31 +8612,34 @@ private VariableComponentSyntax ParseDeconstructionComponent(bool topLevel = fal
                     designation = this.AddError(designation, ErrorCode.ERR_TypeExpected);
                 }
 
-                result = _syntaxFactory.TypedVariableComponent(type, designation);
+                result = _syntaxFactory.DeclarationExpression(type, designation);
             }
 
             return
-                topLevel ? (ComponentHasTypes(result) ? CheckFeatureAvailability(result, MessageID.IDS_FeatureTuples) : null) : result;
+                topLevel ? (TypeFoundInDeconstructionDeclarationVariables(result) ? CheckFeatureAvailability(result, MessageID.IDS_FeatureTuples) : null) : result;
         }
 
-        private static bool ComponentHasTypes(VariableComponentSyntax node)
+        /// <summary>
+        /// Checks if we can find at least one type in the deconstruction variables
+        /// </summary>
+        private static bool TypeFoundInDeconstructionDeclarationVariables(ExpressionSyntax node)
```

Comment:
>TypeFoundInDeconstructionDeclarationVariables [](start = 28, length = 45)

I am not sure if this check is relevant now. Shouldn't we always require Tuples Feature when deconstruction is used?